core/src/sign.c

```diff
@@ -20,18 +20,19 @@
 #include "sign.h"
 #include "memzero.h"
 
-static void compute_prevout_hash(TxInput *inputs, pb_size_t inputs_count,
+static void compute_prevout_hash(const TxInput* const inputs,
+                                 pb_size_t inputs_count,
                                  uint8_t hash[static HASHER_DIGEST_LENGTH]) {
   Hasher hasher;
   hasher_Init(&hasher, HASHER_SHA2D);
   for (int i = 0; i < inputs_count; i++) {
-    TxInput input = inputs[i];
+    const TxInput* const input = &inputs[i];
     DEBUG("Computing prevout hash, input %d", i);
-    hash_rev_bytes(&hasher, input.prev_hash, sizeof(input.prev_hash));
-    print_rev_bytes(input.prev_hash, sizeof(input.prev_hash));
+    hash_rev_bytes(&hasher, input->prev_hash, sizeof(input->prev_hash));
+    print_rev_bytes(input->prev_hash, sizeof(input->prev_hash));
 
-    hash_uint32(&hasher, input.prev_index);
-    print_uint32(input.prev_index);
+    hash_uint32(&hasher, input->prev_index);
+    print_uint32(input->prev_index);
```

Comment:
This is actually a slight functionality change, I changed the code to avoid copying the TxInput to the stack. Happy to put it into a separate PR if needed.